pkg/apis/rke.cattle.io/v1/rkecontrolplane_types.go

```diff
@@ -0,0 +1,180 @@
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// ObservedGeneration is the generation for which the RKEControlPlane has
+	// started processing.
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// CertificateRotationGeneration is the last observed state for which the
+	// certificate rotation operation was successful.
+	// +optional
+	CertificateRotationGeneration int64 `json:"certificateRotationGeneration,omitempty"`
+
+	// RotateEncryptionKeys is the state for which the last encryption key
+	// rotation operation was successful.
+	// +optional
+	RotateEncryptionKeys *RotateEncryptionKeys `json:"rotateEncryptionKeys,omitempty"`
+
+	// RotateEncryptionKeysPhase is the current phase the encryption key
+	// rotation operation is currently executing.
+	// +optional
+	RotateEncryptionKeysPhase RotateEncryptionKeysPhase `json:"rotateEncryptionKeysPhase,omitempty"`
+
+	// RotateEncryptionKeysLeader is the name of the CAPI machine object
+	// which has been elected leader of the controlplane nodes for
+	// encryption key rotation purposes.
+	// +optional
+	RotateEncryptionKeysLeader string `json:"rotateEncryptionKeysLeader,omitempty"`
+
+	// ETCDSnapshotRestore is the state for which the last etcd snapshot
+	// restore operation was successful.
+	// +optional
+	ETCDSnapshotRestore *ETCDSnapshotRestore `json:"etcdSnapshotRestore,omitempty"`
+
+	// ETCDSnapshotRestorePhase is the current phase the etcd snapshot
+	// restore operation is currently executing.
+	// +kubebuilder:validation:Enum=Started;Shutdown;Restore;PostRestorePodCleanup;InitialRestartCluster;PostRestoreNodeCleanup;RestartCluster;Finished;Failed
+	// +optional
+	ETCDSnapshotRestorePhase ETCDSnapshotPhase `json:"etcdSnapshotRestorePhase,omitempty"`
+
+	// ETCDSnapshotCreate is the state for which the last etcd snapshot
+	// create operation was successful.
+	// +optional
+	ETCDSnapshotCreate *ETCDSnapshotCreate `json:"etcdSnapshotCreate,omitempty"`
+
+	// ETCDSnapshotCreatePhase is the current phase the etcd snapshot create
+	// operation is currently executing.
+	// +kubebuilder:validation:Enum=Started;RestartCluster;Finished;Failed
+	// +optional
+	ETCDSnapshotCreatePhase ETCDSnapshotPhase `json:"etcdSnapshotCreatePhase,omitempty"`
+
+	// ConfigGeneration is the current generation of the configuration for a
+	// given cluster.
+	// Changing this value (which is done automatically during an etcd restore) will trigger a reconciliation loop
+	// which will invoke draining (if enabled).
+	// +optional
+	ConfigGeneration int64 `json:"configGeneration,omitempty"`
+
+	// Initialized denotes that the API server is initialized and worker
+	// nodes can be joined to the cluster.
+	// +optional
+	Initialized bool `json:"initialized,omitempty"`
+
+	// AgentConnected denotes that the cluster-agent connection is currently
+	// established for the cluster.
+	// +optional
+	AgentConnected bool `json:"agentConnected,omitempty"`
+}
+
+// +genclient
+// +kubebuilder:object:root=true
```

Comment:
Do we need this marker? I think it's only used to generate some other kubebuilder go objects, not the crd. I tried removing it and re-generating, and it still worked.

Reply:
Oh if we don't need these I'll remove them.